pkg/scan/scan.go

```diff
@@ -50,48 +49,57 @@ func printSummary(result components.HealthStateCheckResult) {
 
 // Runs the scan operations.
 func Scan(ctx context.Context, opts ...OpOption) error {
-	defaultCheckHealthStateFuncs := []func(context.Context) (components.HealthStateCheckResult, error){
-		oscomponent.CheckHealthState,
-		cpucomponent.CheckHealthState,
-		memorycomponent.CheckHealthState,
-	}
 
 	op := &Op{}
 	if err := op.applyOpts(opts); err != nil {
 		return err
 	}
-	if op.netcheck {
-		defaultCheckHealthStateFuncs = append(defaultCheckHealthStateFuncs, networklatencycomponent.CheckHealthState)
-	}
 
 	fmt.Printf("\n\n%s scanning the host (GOOS %s)\n\n", inProgress, runtime.GOOS)
 
-	for _, checkFunc := range defaultCheckHealthStateFuncs {
-		rs, err := checkFunc(ctx)
-		if err != nil {
-			return err
-		}
-		printSummary(rs)
-	}
-
 	nvidiaInstalled, err := nvidia_query.GPUsInstalled(ctx)
 	if err != nil {
 		log.Logger.Warnw("error checking nvidia gpu installation", "error", err)
 		return err
 	}
 
+	var nvmlInstance nvidianvml.InstanceV2
 	if nvidiaInstalled {
 		fmt.Printf("\n%s scanning nvidia accelerators\n", inProgress)
-
-		nvmlInstance, err := nvidianvml.NewInstanceV2()
+		nvmlInstance, err = nvidianvml.NewInstanceV2()
 		if err != nil {
 			return err
 		}
-		nvidiaTempData, err := nvidiatemperaturecomponent.CheckHealthState(nvmlInstance)
+	}
+
+	gpudInstance := components.GPUdInstance{
+		RootCtx: ctx,
+
+		NVMLInstance: nvmlInstance,
+		NVIDIAToolOverwrites: nvidiacommon.ToolOverwrites{
+			IbstatCommand: op.ibstatCommand,
+		},
+
+		EventStore:       nil,
+		RebootEventStore: nil,
+	}
+
+	componentInits := []components.InitFunc{
+		oscomponent.New,
+	}
+
+	for _, initFunc := range componentInits {
+		c, err := initFunc(gpudInstance)
 		if err != nil {
 			return err
 		}
-		printSummary(nvidiaTempData)
+		defer c.Close()
```

Comment:
This is not a good practice. The for loop is short and we can close it after the check.